frontend/src/components/DetailsTable.tsx

```diff
@@ -49,62 +50,82 @@ export const css = stylesheet({
 });
 
 interface DetailsTableProps {
-  fields: Array<KeyValue<string | S3Artifact>>;
+  fields: Array<KeyValue<string | Partial<S3Artifact>>>;
   title?: string;
-  valueComponent?: React.FC<S3Artifact>;
+}
+
+interface DetailsFieldValueProps {
+  index?: number;
+  fieldname?: string;
+  fieldvalue?: string | Partial<S3Artifact>;
 }
 
 function isString(x: any): x is string {
   return typeof x === 'string';
 }
 
+export const DetailsFieldValue: React.FC<DetailsFieldValueProps> = ({
+  index,
+  fieldname,
+  fieldvalue,
+}) => {
+  // either return the plain text, or convert fieldvalue to a js obj
+  if (isString(fieldvalue)) {
+    try {
+      const parsedJson = JSON.parse(fieldvalue);
+      // Nulls, booleans, strings, and numbers can all be parsed as JSON, but we don't care
+      // about rendering. Note that `typeOf null` returns 'object'
+      if (parsedJson === null || typeof parsedJson !== 'object') {
+        throw new Error('Parsed JSON was neither an array nor an object. Using default renderer');
+      }
+      // set the fieldvalue to be a js object.
+      fieldvalue = parsedJson;
+    } catch (error) {
+      // returns a simple text if the string is not a js object
+      return <span>{fieldvalue}</span>;
+    }
+  }
+
+  // if fieldvalue is an argo s3 artifact, renders a preview.
+  if (isS3Artifact(fieldvalue)) {
+    return <MinioArtifactPreview artifact={fieldvalue} />;
+  }
+
+  // if js object, renders a editor
+  if (fieldvalue && typeof fieldvalue === 'object') {
+    return (
+      <div key={index} className={css.row}>
```

Comment:
```suggestion
      <div className={css.row}>
```
You don't need a key here, it's not a list.

Reply:
Hmm, actually, why do we need the prop index?

Reply:
Actually, I have no idea too. It was in the original code, I kept it inside in case of breaking anything thou. 

I reordered the rendering sequence to support the preview.